### PR TITLE
Revamp BaseSingleAxisSubsystem

### PR DESCRIPTION
**Pull request by having11**

### CODE REVIEW

File: src/main/include/utils/PidMotorController.h [hunk 1]
```diff
@@ -1,53 +1,247 @@
...
+
+  void SetAbsoluteEncoderConversionFactor(double factor) {
+    if (m_absEncoder) {
+      m_absEncoder->SetPositionConversionFactor(factor);
+    }
+  }
 
   /// @brief Stop the motor
-  void Stop();
+  // TODO: USE MOTOR HERE, THIS IS BAD AND SHOULD NOT BE EMPTY
```
Comment (nobody5050) [inline src/main/include/utils/PidMotorController.h:136]: Can this todo be removed?

File: src/main/include/subsystems/WristSubsystem.h [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  void Periodic() override {
+    RotationalSingleAxisSubsystem<rev::CANSparkMax, rev::SparkPIDController,
+                                  rev::SparkRelativeEncoder,
+                                  rev::SparkAbsoluteEncoder>::Periodic();
+
+    // armTuner.UpdateFromShuffleboard();
+  }
+
+ private:
+  // :((((
```
Comment (nobody5050) [inline src/main/include/subsystems/WristSubsystem.h:54]: :((((

**Review by nobody5050 — COMMENTED** [review]
Looks very clean, do we want to go ahead with renaming the [class]2 to just [class] before merge or are we waiting until climbers can be converted?

File: src/main/include/utils/PidMotorController.h (hunk 1 above)
Comment (having11) [reply src/main/include/utils/PidMotorController.h:136]: ye

**Comment by having11** [comment]
> Looks very clean, do we want to go ahead with renaming the [class]2 to just [class] before merge or are we waiting until climbers can be converted?

I would convert and check for regressions before deleting existing code

**Review by 5690Programmers — APPROVED** [review]
Very nice, Evan!

**Pull request merged**